stdlib/Dates/test/io.jl

```diff
@@ -545,4 +554,26 @@ end
     @test Time("13:24", "II:MMp") == Time("13:24", "HH:MM")
 end
 
+@testset "Issue #10561, two-digit year parsing ambiguities" begin
+    error_dates = Dict()
+    expected_error_dates = Dict()
+
+    for test_year in 0:1:3000
```

Comment:
We don't usually run tests that cover all possible values as their accumulation would take a lot of time on CI. Better test a few dates illustrating the different cases, notably corner or limit cases.

Reply:
Thanks for the feedback. Pushed an update. I'll squash this once you're happy.